src/status_im2/subs/contact_test.cljs

```diff
@@ -0,0 +1,185 @@
+              :Removed               false
+              :trustStatus           0
+              :alias                 "Instant noodles"
+              :bio                   ""
+              :IsSyncing             false
+              :display-name          ""
+              :ens-verified          true
+              :socialLinks           nil
+              :blocked               false
+              :allow-new-users?      true
+              :verificationStatus    0
+              :lastUpdatedLocally    1672582563204
+              :public-key            "0xtest"
+              :names                 {:nickname         nil
+                                      :display-name     ""
+                                      :three-words-name "Instant noodles"
+                                      :ens-name         "@slim.shady"}
+              :has-added-us          true
+              :contact-request-state 1})}
+    {:title "R"
+     :data  ({:active?               true
+              :last-updated          1672582629695
+              :mutual?               true
+              :blocked?              false
+              :contactRequestClock   0
+              :images                {}
+              :added                 true
+              :name                  "slim.shady"
+              :added?                true
+              :Removed               false
+              :trustStatus           0
+              :alias                 "Real Slim Shady"
+              :bio                   ""
+              :IsSyncing             false
+              :display-name          ""
+              :ens-verified          true
+              :socialLinks           nil
+              :blocked               false
+              :allow-new-users?      true
+              :verificationStatus    0
+              :lastUpdatedLocally    1672582563204
+              :public-key            "0xtest"
+              :names                 {:nickname         nil
+                                      :display-name     ""
+                                      :three-words-name "Real Slim Shady"
+                                      :ens-name         "@slim.shady"}
+              :has-added-us          true
+              :contact-request-state 1})}))
+
+(h/deftest-sub :contacts/sorted-and-grouped-by-first-letter
+  [sub-name]
+  (testing "Returning empty sequence when no contacts"
+    (swap! rf-db/app-db merge (assoc contacts-sample-data :contacts/contacts {}))
+    (is (empty? (rf/sub [sub-name]))))
+  (testing "Returning empty sequence when no mutual contacts"
+    (let [remove-contact-as-mutual #(-> %
+                                        (assoc-in ["0xtest" :mutual?] false)
+                                        (assoc-in ["0xtest2" :mutual?] false)
+                                        (assoc-in ["0xtest3" :mutual?] false))]
+      ;; TODO(@ibrkhalil,2023-01-01): Replace with update-vals when we update the ClojureScript API
```

Comment:
`update-vals` is a great addition to Clojure 1.11, but I think using it doesn't deserve a TODO in our code. Your solution using `swap!` and `update` is just fine.